### ISSUE TITLE
Infinite gradients in training log

### ISSUE THREAD

**Issue by clane9** [MedARC-AI/fmri-fm#6]
The pretraining log often reports infinite gradients:

```
Train: [2]  [5500/6250]  eta: 0:01:18  lr: 0.000072  grad: 0.0877 (inf)  loss: 0.9832 (0.9838)  time: 0.0967  data: 0.0219  max mem: 4764
```

Presumably these are due to infrequent gradient spikes, resulting in an `inf` moving average. Still, it could be worth investigating why exactly this is happening and whether some small change to the [logger](https://github.com/MedARC-AI/fmri-fm/blob/b3d97ca3b5bd5ce003628236811136d56c7f4ece/src/flat_mae/utils.py#L47-L50) will fix it.
